src/types.js

```diff
@@ -276,27 +276,43 @@ export type Grip = {
  * @static
  */
 
-export type Source = {|
+export type JsSource = {|
   +id: string,
   +url: string,
   +sourceMapURL?: string,
   +isBlackBoxed: boolean,
   +isPrettyPrinted: boolean,
-  +isWasm: boolean,
+  +isWasm: false,
   +text?: string,
   +contentType?: string,
   +error?: string,
   +loadedState: "unloaded" | "loading" | "loaded",
   +relativeUrl?: string
 |};
 
+export type WasmSource = {|
+  +id: string,
+  +url: string,
+  +sourceMapURL?: string,
+  +isBlackBoxed: boolean,
+  +isPrettyPrinted: boolean,
+  +isWasm: true,
+  +text?: {| binary: Object |},
+  +contentType?: string,
+  +error?: string,
+  +loadedState: "unloaded" | "loading" | "loaded",
+  +relativeUrl?: string
+|};
```

Comment:
Is there a way we can create a base type and copy from it since most of the properties and value types are the same?

Reply:
Yes, we totally can